### ISSUE TITLE
hydra.errors.InstantiationException: Error locating target 'diffcsp.pl_data.datamodule.CrystDataModule', set env var HYDRA_FULL_ERROR=1 to see chained exception. full_key: data.datamodule

### ISSUE THREAD

**Issue by yuuhaixia** [jiaor17/DiffCSP#14]
Thank you very much for your contribution, do you know how to fix the following error?

/data/run01/scw6cse/DiffCSP-main/diffcsp/run.py:172: UserWarning: 
The version_base parameter is not specified.
Please specify a compatability version level, or None.
Will assume defaults for version 1.1
  @hydra.main(config_path=str(PROJECT_ROOT / "conf"), config_name="default")
/HOME/scw6cse/.conda/envs/py311/lib/python3.11/site-packages/hydra/_internal/defaults_list.py:251: UserWarning: In 'default': Defaults list is missing `_self_`. See https://hydra.cc/docs/1.2/upgrades/1.0_to_1.1/default_composition_order for more information
  warnings.warn(msg, UserWarning)
/HOME/scw6cse/.conda/envs/py311/lib/python3.11/site-packages/hydra/_internal/hydra.py:119: UserWarning: Future Hydra versions will no longer change working directory at job runtime by default.
See https://hydra.cc/docs/1.2/upgrades/1.1_to_1.2/changes_to_job_working_dir/ for more information.
  ret = run_job(
[rank: 0] Seed set to 42
[2024-05-14 10:07:24,302][hydra.utils][INFO] - Instantiating <diffcsp.pl_data.datamodule.CrystDataModule>
Error executing job with overrides: ['data=mp_20', 'expname=mp_20_result']

-------------------------------------------------------------------------------
run.py 178 <module>
main()

main.py 94 decorated_main
_run_hydra(

utils.py 394 _run_hydra
_run_app(

utils.py 457 _run_app
run_and_report(

utils.py 223 run_and_report
raise ex

utils.py 220 run_and_report
return func()

utils.py 458 <lambda>
lambda: hydra.run(

hydra.py 132 run
_ = ret.return_value

utils.py 260 return_value
raise self._return_value

utils.py 186 run_job
ret.return_value = task_function(task_cfg)

run.py 174 main
run(cfg)

run.py 93 run
datamodule: pl.LightningDataModule = hydra.utils.instantiate(

_instantiate2.py 226 instantiate
return instantiate_node(

_instantiate2.py 333 instantiate_node
_target_ = _resolve_target(node.get(_Keys.TARGET), full_key)

_instantiate2.py 139 _resolve_target
raise InstantiationException(msg) from e

hydra.errors.InstantiationException:
Error locating target 'diffcsp.pl_data.datamodule.CrystDataModule', set env var HYDRA_FULL_ERROR=1 to see chained exception.
full_key: data.datamodule

Here is my environment：

aiohttp                  3.9.5
aiosignal                1.3.1
antlr4-python3-runtime   4.9.3
ase                      3.22.1
attrs                    23.2.0
certifi                  2024.2.2
charset-normalizer       3.3.2
click                    8.1.7
cmake                    3.29.3
colorama                 0.4.6
contourpy                1.2.1
cycler                   0.12.1
dill                     0.3.8
docker-pycreds           0.4.0
filelock                 3.14.0
fonttools                4.51.0
frozenlist               1.4.1
fsspec                   2024.3.1
future                   0.18.3
gitdb                    4.0.11
GitPython                3.1.43
hydra-core               1.3.2
idna                     3.7
Jinja2                   3.1.4
joblib                   1.3.2
kiwisolver               1.4.5
latexcodec               2.0.1
lightning-utilities      0.11.2
lit                      18.1.4
MarkupSafe               2.1.5
matplotlib               3.8.4
monty                    2023.11.3
mpmath                   1.3.0
multidict                6.0.5
multiprocess             0.70.16
networkx                 3.1
numpy                    1.26.4
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.2.10.91
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusparse-cu11     11.7.4.91
nvidia-nccl-cu11         2.14.3
nvidia-nvtx-cu11         11.7.91
omegaconf                2.3.0
p-tqdm                   1.4.0
packaging                24.0
palettable               3.3.3
pandas                   2.2.2
pathos                   0.3.2
Pillow                   10.0.1
pip                      24.0
platformdirs             4.2.1
plotly                   5.18.0
pox                      0.3.4
ppft                     1.7.6.8
pretty-errors            1.2.25
protobuf                 4.25.3
psutil                   5.9.6
pybtex                   0.24.0
pymatgen                 2023.11.12
pyparsing                3.1.2
python-dateutil          2.9.0.post0
python-dotenv            1.0.1
pytorch-lightning        2.2.4
pytz                     2024.1
PyYAML                   6.0.1
requests                 2.31.0
ruamel.yaml              0.18.6
ruamel.yaml.clib         0.2.8
scikit-learn             1.3.2
scipy                    1.13.0
sentry-sdk               2.1.1
setproctitle             1.3.3
setuptools               68.0.0
six                      1.16.0
SMACT                    2.5.4
smmap                    5.0.1
spglib                   2.1.0
sympy                    1.12
tabulate                 0.9.0
tenacity                 8.2.3
threadpoolctl            3.2.0
torch                    2.0.0
torch-cluster            1.6.3+pt20cu118
torch_geometric          2.4.0
torch-scatter            2.1.2+pt20cu118
torch-sparse             0.6.18+pt20cu118
torch-spline-conv        1.2.2+pt20cu118
torchaudio               2.0.1
torchmetrics             1.4.0
torchvision              0.15.1
tqdm                     4.66.1
triton                   2.0.0
typing_extensions        4.11.0
tzdata                   2024.1
uncertainties            3.1.7
urllib3                  2.2.1
wandb                    0.17.0
wheel                    0.43.0
yarl                     1.9.4

**Comment by oridwan** [comment]
@yuuhaixia @jiaor17  have you solved it?  I am facing the sam problem

**Comment by mzj-ustc** [comment]
me too. Have you solved this problem?

**Comment by zq-meng** [comment]
Have you solved this problem?

**Comment by freedom0625** [comment]
这个是?



---- Replied Message ----
| From | ***@***.***> |
| Date | 02/13/2025 22:53 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | Re: [jiaor17/DiffCSP] hydra.errors.InstantiationException: Error locating target 'diffcsp.pl_data.datamodule.CrystDataModule', set env var HYDRA_FULL_ERROR=1 to see chained exception. full_key: data.datamodule (Issue #14) |

Have you solved this problem?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

zq-meng left a comment (jiaor17/DiffCSP#14)

Have you solved this problem?

**Comment by ZhixunLEE** [comment]
You can enable HYDRA_FULL_ERROR=1 to see the full error traceback.

Then you can see detailed error messages. It could be because you haven't installed all the necessary packages properly (at least that was the case for me).
